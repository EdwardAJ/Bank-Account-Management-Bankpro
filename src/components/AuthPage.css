.container-auth {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: 'Open Sans', sans-serif;
}

#login-header {
  font-size: 0.75rem;
  color: white;
  font-weight: 400;
  text-align: left;
  letter-spacing: 0.02rem;
  margin-left: 0.5%;
}

.submit {
  margin-top: 17.5px;
  letter-spacing: 0.05rem;
  border: solid white;
  border-width: 1px;
  background-color: #005b69;
  width: 270px;
  height: 42.5px;
  border-radius: 8px;
  color: white;
  cursor: pointer;
}

.form {
    border: none;
    width: 252.5px;
    height: 37.5px;
    border-radius: 8px;
    font-size: 1rem;
    letter-spacing: 0.05rem;
    padding-left: 17.5px;
    color: #005b69;
  }