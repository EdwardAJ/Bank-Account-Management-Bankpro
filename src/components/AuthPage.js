import React from 'react';
import './AuthPage.css';

import axios from 'axios';
import { LOGIN_URL } from './../utils/endpoints.js';
import { HEADERS } from './../utils/header.js';
import getXMLResponse from './../utils/functions.js';
import { createCookie } from './../utils/cookie.js';

export default class AuthPage extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            value: ''
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange (event) {
        this.setState({value: event.target.value});
    }

    handleSubmit (event) {
        let requestBody =
        '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://controllers.webservice.com/">' +
            '<soapenv:Header/>' +
            '<soapenv:Body>' +
                '<con:validateAccountNumber>' +
                    '<accountNumber>' +  this.state.value + '</accountNumber>' +
                '</con:validateAccountNumber>' +
            '</soapenv:Body>' +
        '</soapenv:Envelope>';

        axios.post(LOGIN_URL, requestBody, {
            headers: HEADERS
        }).then((response) => {
            if (response.status === 200) {
                let resultStr = getXMLResponse(response.data)[0].innerHTML;
                if (resultStr === 'true') {
                    createCookie("auth", this.state.value, 1);
                    this.props.handleLogin();
                } else {
                    alert("Silakan masukkan nomor yang valid.");
                }
            }
        }).catch((error) => {
            console.log("Error: ", error);
        })
        event.preventDefault();
    }
    
    render() {
        return (
            <div className="container">
                <div>
                    <p id="login-header"> Masukkan Nomor Rekening </p>
                    <div className="form-area">
                        <form>
                            <input className="form" type="text" value={ this.state.value } onChange={ this.handleChange } placeholder="Example: 13517070" />
                        </form> 
                    </div>
                    <form>
                        <button className="submit" onClick={ this.handleSubmit }>LOGIN</button>
                    </form>
                </div>
            </div>
        );
    }
}

