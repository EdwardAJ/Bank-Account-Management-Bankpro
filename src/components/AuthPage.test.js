import React from 'react';
import { shallow } from 'enzyme';
import AuthPage from './AuthPage';

describe("AuthPage", () => {
    it("should render my component", () => {
        const component = shallow(<AuthPage />);
        expect(component.getElements()).toMatchSnapshot();
    });
});

describe("AuthPage", () => {
    it("should render this component", () => {
        const component = shallow(<AuthPage />);
        expect(component.find('#login-header').text()).toEqual(" Masukkan Nomor Rekening ");
        expect(component.exists('.form')).toEqual(true);
    });
});