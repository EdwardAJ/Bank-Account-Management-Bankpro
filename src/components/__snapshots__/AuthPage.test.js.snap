exports[`AuthPage should render my component 1`] = `
Array [
  <div
    className="container-auth">
    <div>
        <p
            id="login-header">
             Masukkan Nomor Rekening 
        </p>
        <div
            className="form-area">
            <form>
                <input
                    className="form"
                    onChange={[Function]}
                    placeholder="Example: 13517070"
                    type="text"
                    value="" />
            </form>
        </div>
        <form>
            <button
                className="submit"
                onClick={[Function]}>
                LOGIN
            </button>
        </form>
    </div>
</div>,
]
`;